libs/language-server/src/lib/validation/validation-util.ts

```diff
@@ -135,6 +136,39 @@ function collectSubExpressionsWithoutFreeVariables(
       return [expression];
     }
     return [...leftSubExpressions, ...rightSubExpressions];
+  } else if (isTernaryExpression(expression)) {
+    const firstExpression = expression.first;
+    const secondExpression = expression.second;
+    const thirdExpression = expression.third;
```

Comment:
All logic after extracting the sub-expressions from the expression is the same as for binary expression... I think refactoring this into a shared logic would make sense and make this easier to read. Again, imho something for a follow up task.